--- a/ajax/getSettings.php
+++ b/ajax/getSettings.php
@@ -45,8 +45,17 @@
 $data ['screenMediaExtension']['firefox'] = trim($config->getAppValue('ojsxc', 'firefoxExtension'));
 $data ['screenMediaExtension']['chrome'] = trim($config->getAppValue('ojsxc', 'chromeExtension'));
 
-$data ['xmpp'] ['url'] = trim($config->getAppValue('ojsxc', 'boshUrl'));
-$data ['xmpp'] ['domain'] = trim($config->getAppValue('ojsxc', 'xmppDomain'));
+$request = \OC::$server->getRequest();
+$boshUrl = trim($config->getAppValue('ojsxc', 'boshUrl'));
+if (empty($boshUrl) && $request) {
+  $boshUrl = 'https://' . $request->getServerHost() . '/http-bind/';
+}
+$data ['xmpp'] ['url'] = $boshUrl;
+$xmppDomain = trim($config->getAppValue('ojsxc', 'xmppDomain'));
+if (empty($xmppDomain)) {
+  $xmppDomain = $request->getServerHost();
+}
+$data ['xmpp'] ['domain'] = $xmppDomain;
 $data ['xmpp'] ['resource'] = trim($config->getAppValue('ojsxc', 'xmppResource'));
 $data ['xmpp'] ['overwrite'] = validateBoolean($config->getAppValue('ojsxc', 'xmppOverwrite'));
 $data ['xmpp'] ['onlogin'] = null;
--- a/settings.php
+++ b/settings.php
@@ -7,9 +7,22 @@
 $config = \OC::$server->getConfig();
 $tmpl = new OCP\Template('ojsxc', 'settings');
 
-$tmpl->assign('serverType', $config->getAppValue('ojsxc', 'serverType'));
-$tmpl->assign('boshUrl', $config->getAppValue('ojsxc', 'boshUrl'));
-$tmpl->assign('xmppDomain', $config->getAppValue('ojsxc', 'xmppDomain'));
+$serverType = $config->getAppValue('ojsxc', 'serverType');
+$request = \OC::$server->getRequest();
+$boshUrl = $config->getAppValue('ojsxc', 'boshUrl');
+if (empty($boshUrl) && $request) {
+  $boshUrl = 'https://' . $request->getServerHost() . '/http-bind/';
+}
+$tmpl->assign('boshUrl', $boshUrl);
+$xmppDomain = $config->getAppValue('ojsxc', 'xmppDomain');
+if (empty($xmppDomain)) {
+  $xmppDomain = $request->getServerHost();
+}
+if (empty($serverType) && !empty($boshUrl) && !empty($xmppDomain)) {
+  $serverType = 'external';
+}
+$tmpl->assign('serverType', $serverType);
+$tmpl->assign('xmppDomain', $xmppDomain);
 $tmpl->assign('xmppPreferMail', $config->getAppValue('ojsxc', 'xmppPreferMail'));
 $tmpl->assign('xmppResource', $config->getAppValue('ojsxc', 'xmppResource'));
 $tmpl->assign('xmppOverwrite', $config->getAppValue('ojsxc', 'xmppOverwrite'));
