--- a/lib/Controller/SettingsController.php
+++ b/lib/Controller/SettingsController.php
@@ -70,16 +70,25 @@
 			'chrome' => trim($this->getAppValue('chromeExtension'))
 		 ];
 
+		$request = \OC::$server->getRequest();
+		$boshUrl = trim($this->getAppValue('boshUrl'));
+		if (empty($boshUrl) && $request) {
+			$boshUrl = 'https://' . $request->getServerHost() . '/http-bind/';
+		}
+		$xmppDomain = trim($this->getAppValue('xmppDomain'));
+		if (empty($xmppDomain)) {
+			$xmppDomain = $request->getServerHost();
+		}
 		$data ['xmpp'] = [
-			'url' => trim($this->getAppValue('boshUrl')),
-			'domain' => trim($this->getAppValue('xmppDomain')),
+			'url' => $boshUrl,
+			'domain' => $xmppDomain,
 			'resource' => trim($this->getAppValue('xmppResource')),
 			'overwrite' => $this->getBooleanAppValue('xmppOverwrite'),
 			'onlogin' => null
 		 ];
 
 		$data ['adminSettings'] = [
-			'xmppDomain' => trim($this->getAppValue('xmppDomain'))
+			'xmppDomain' => $xmppDomain
 		 ];
 
 		if ($this->getBooleanAppValue('xmppPreferMail') && !$this->getBooleanAppValue('timeLimitedToken')) {
--- a/lib/Settings/Admin.php
+++ b/lib/Settings/Admin.php
@@ -28,10 +28,22 @@
 
 		$apiUrl = \OC::$server->getURLGenerator()->linkToRouteAbsolute('ojsxc.externalApi.index');
 
+		$request = \OC::$server->getRequest();
+		$boshUrl = $this->config->getAppValue('ojsxc', 'boshUrl');
+		if (empty($boshUrl) && $request) {
+			$boshUrl = 'https://' . $request->getServerHost() . '/http-bind/';
+		}
+		$xmppDomain = $this->config->getAppValue('ojsxc', 'xmppDomain');
+		if (empty($xmppDomain)) {
+			$xmppDomain = $request->getServerHost();
+		}
+		if (empty($serverType) && !empty($boshUrl) && !empty($xmppDomain)) {
+			$serverType = 'external';
+		}
 		$parameters = [
 		   'serverType' => (!empty($serverType))? $serverType : 'internal',
-		   'boshUrl' => $this->config->getAppValue('ojsxc', 'boshUrl'),
-		   'xmppDomain' => $this->config->getAppValue('ojsxc', 'xmppDomain'),
+		   'boshUrl' => $boshUrl,
+		   'xmppDomain' => $xmppDomain,
 		   'xmppPreferMail' => $this->config->getAppValue('ojsxc', 'xmppPreferMail'),
 		   'xmppResource' => $this->config->getAppValue('ojsxc', 'xmppResource'),
 		   'xmppOverwrite' => $this->config->getAppValue('ojsxc', 'xmppOverwrite'),
