--- a/js/jsxc/jsxc.js
+++ b/js/jsxc/jsxc.js
@@ -5757,7 +5757,7 @@
    var httpUploadOptions = jsxc.options.get('httpUpload') || {};
    var maxSize = httpUploadOptions.maxSize || 0;
 
-   if (file.transportMethod !== 'webrtc' && jsxc.xmpp.httpUpload.ready && maxSize >= 0 && file.size > maxSize) {
+   if (file.transportMethod !== 'webrtc' && jsxc.xmpp.httpUpload.ready && maxSize && file.size > maxSize) {
       jsxc.debug('File too large for http upload.');
 
       if (jsxc.fileTransfer.isWebrtcCapable(bid)) {
