--- a/js/jsxc/jsxc.js
+++ b/js/jsxc/jsxc.js
@@ -5755,7 +5755,7 @@
 jsxc.fileTransfer.fileSelected = function(jid, msg, file) {
    var bid = jsxc.jidToBid(jid);
    var httpUploadOptions = jsxc.options.get('httpUpload') || {};
-   var maxSize = httpUploadOptions.maxSize || 0;
+   var maxSize = httpUploadOptions.maxSize || -1;
 
    if (file.transportMethod !== 'webrtc' && jsxc.xmpp.httpUpload.ready && maxSize >= 0 && file.size > maxSize) {
       jsxc.debug('File too large for http upload.');
