From 9d3146f0eb0ad9e14dd7cdc2a5414fc157bb8eb7 Mon Sep 17 00:00:00 2001
From: sualko <klaus@jsxc.org>
Date: Tue, 25 Apr 2017 14:06:39 +0200
Subject: [PATCH] fix log in after connection failure (fix jsxc/jsxc#516)

---
 js/ojsxc.js | 8 ++++++++
 1 file changed, 8 insertions(+)

--- a/js/ojsxc.js
+++ b/js/ojsxc.js
@@ -338,5 +338,13 @@
 
       var alt = $('<p id="jsxc_alt"/>').append(link);
       $('#body-login form:eq(0) fieldset').append(alt);
+
+      Strophe.log = function(level, msg) {
+         if (level === 3 && /^request id/.test(msg)) {
+            console.warn('Something went wrong during BOSH connection establishment. Continue without chat.');
+
+            jsxc.submitLoginForm();
+         }
+      };
    }
 });
