--- a/auth_ejabberd_oc.php
+++ b/auth_ejabberd_oc.php
@@ -26,7 +26,7 @@
 //#############Start#of#Config####################
 
 /** The root of your owncloud installation. */
-$oc_root = '/var/www/owncloud/';
+$oc_root = '/usr/lib/nextcloud/www/';
 
 /** Should we log the activity of this script? */
 $debugging = true;
@@ -34,7 +34,7 @@
 //#############End#of#config######################
 
 error_reporting(E_ALL);
-ini_set('error_log', '/var/log/ejabberd/oc.log');
+ini_set('error_log', '/var/log/ejabberd/nc.log');
 
 $auth = new JabberAuth();
 $auth->play(); // We simply start process !
@@ -184,7 +184,7 @@
 		$user = escapeshellarg($this->jabber_user);
 		$pass = escapeshellarg($this->jabber_pass);
 
-		$ret = exec("/etc/ejabberd/auth_oc_user.php '$oc_root' $user $pass");
+		$ret = exec("/usr/share/nextcloud-app-ojsxc/auth_oc_user.php '$oc_root' $user $pass");
 		
 		$this->logg("Password is " . ($ret == '1'));
 
@@ -201,7 +201,7 @@
 
 		$user = escapeshellarg($this->jabber_user);
 
-		$ret = exec("/etc/ejabberd/auth_oc_user.php '$oc_root' $user");
+		$ret = exec("/usr/share/nextcloud-app-ojsxc/auth_oc_user.php '$oc_root' $user");
 
 		return $ret == '1';
 	}
