--- a/templates/settings/admin.php
+++ b/templates/settings/admin.php
@@ -13,6 +13,13 @@
 		<h3>Server type</h3>
 		<div class="form-group">
 			<label class="text-left form-no-padding">
+				<input type="radio" name="serverType" class="required" required="required" value="spreedbox" <?php if($_['serverType'] === 'spreedbox')echo 'checked'; ?> />
+				Spreedbox
+			</label>
+			<em>Choose this option to use the XMPP server of the Spreedbox.</em>
+		</div>
+		<div class="form-group">
+			<label class="text-left form-no-padding">
 				<input type="radio" name="serverType" required="required" value="internal" <?php if($_['serverType'] === 'internal')echo 'checked'; ?> />
 				Internal
 			</label>
@@ -39,8 +46,8 @@
 
 			</div>
 
-			<div class="ojsxc-external hidden">
-				<div class="form-group">
+			<div class="ojsxc-external ojsxc-spreedbox hidden">
+				<div class="form-group ojsxc-external">
 					<label for="xmppDomain">* XMPP domain</label>
 					<div class="form-col">
 						<input type="text" name="xmppDomain" id="xmppDomain" class="required" required="required" value="<?php p($_['xmppDomain']); ?>" />
@@ -50,7 +57,7 @@
 					<label for="xmppPreferMail">Prefer mail address to loginName@xmppDomain</label>
 					<input type="checkbox" name="xmppPreferMail" id="xmppPreferMail" value="true" <?php if($_[ 'xmppPreferMail']==='true' || $_[ 'xmppPreferMail']===true) echo "checked"; ?> />
 				</div>
-				<div class="form-group">
+				<div class="form-group ojsxc-external">
 					<label for="boshUrl">* BOSH url</label>
 					<div class="form-col">
 						<input type="text" name="boshUrl" id="boshUrl" class="required" required="required" value="<?php p($_['boshUrl']); ?>" />
@@ -154,7 +161,7 @@
 			</div>
 		</fieldset>
 
-		<fieldset>
+		<fieldset class="ojsxc-external">
 			<h3>ICE server <small>(WebRTC)</small></h3>
 			<div class="form-group">
 				<label for="iceUrl">URLs</label>
@@ -192,7 +199,7 @@
 				</div>
 			</div>
 		</fieldset>
-		<fieldset>
+		<fieldset class="ojsxc-external">
 			<h3>Screen sharing</h3>
 			<div class="form-group">
 				<label for="firefoxExtension">Firefox Extension URL</label>
@@ -209,7 +216,7 @@
 				</div>
 			</div>
 		</fieldset>
-		<fieldset>
+		<fieldset class="ojsxc-external">
 			<h3>CSP <small>Content-Security-Policy</small></h3>
 			<div class="form-group">
 				<label for="fileUpload">External services</label>
--- a/js/settings/admin.js
+++ b/js/settings/admin.js
@@ -94,10 +94,13 @@
    }
 
    $('#ojsxc [name=serverType]').change(function() {
-      $('#ojsxc .ojsxc-external, #ojsxc .ojsxc-internal, #ojsxc .ojsxc-managed').hide();
-      $('#ojsxc .ojsxc-external, #ojsxc .ojsxc-internal, #ojsxc .ojsxc-managed').find('.required').removeAttr('required');
+      $('#ojsxc .ojsxc-external, #ojsxc .ojsxc-internal, #ojsxc .ojsxc-managed, #ojsxc .ojsxc-spreedbox').hide();
+      $('#ojsxc .ojsxc-external, #ojsxc .ojsxc-internal, #ojsxc .ojsxc-managed, #ojsxc .ojsxc-spreedbox').find('.required').removeAttr('required');
       $('#ojsxc .ojsxc-' + $(this).val()).show();
       $('#ojsxc .ojsxc-' + $(this).val()).find('.required').attr('required', 'true');
+      if ($(this).val() === 'spreedbox') {
+         $('#ojsxc .form-group.ojsxc-external').find('.required').removeAttr('required');
+      }
    });
    $('#ojsxc [name=serverType]:checked').change();
 
--- a/lib/Settings/Admin.php
+++ b/lib/Settings/Admin.php
@@ -24,26 +24,14 @@
 		$externalServices = $this->config->getAppValue('ojsxc', 'externalServices');
 		$externalServices = explode("|", $externalServices);
 
-		$serverType = $this->config->getAppValue('ojsxc', 'serverType');
+		$serverType = $this->config->getAppValue('ojsxc', 'serverType', 'spreedbox');
 
 		$apiUrl = \OC::$server->getURLGenerator()->linkToRouteAbsolute('ojsxc.externalApi.index');
 
-		$request = \OC::$server->getRequest();
-		$boshUrl = $this->config->getAppValue('ojsxc', 'boshUrl');
-		if (empty($boshUrl) && $request) {
-			$boshUrl = 'https://' . $request->getServerHost() . '/http-bind/';
-		}
-		$xmppDomain = $this->config->getAppValue('ojsxc', 'xmppDomain');
-		if (empty($xmppDomain)) {
-			$xmppDomain = $request->getServerHost();
-		}
-		if (empty($serverType) && !empty($boshUrl) && !empty($xmppDomain)) {
-			$serverType = 'external';
-		}
 		$parameters = [
 		   'serverType' => (!empty($serverType))? $serverType : 'internal',
-		   'boshUrl' => $boshUrl,
-		   'xmppDomain' => $xmppDomain,
+		   'boshUrl' => $this->config->getAppValue('ojsxc', 'boshUrl'),
+		   'xmppDomain' => $this->config->getAppValue('ojsxc', 'xmppDomain'),
 		   'xmppPreferMail' => $this->config->getAppValue('ojsxc', 'xmppPreferMail'),
 		   'xmppResource' => $this->config->getAppValue('ojsxc', 'xmppResource'),
 		   'xmppOverwrite' => $this->config->getAppValue('ojsxc', 'xmppOverwrite'),
--- a/lib/ContactsMenu/Providers/ChatProvider.php
+++ b/lib/ContactsMenu/Providers/ChatProvider.php
@@ -54,7 +54,7 @@
 			$xmppPreferMail = $xmppPreferMail === true || $xmppPreferMail === 'true';
 			$serverType = $config->getAppValue('ojsxc', 'serverType', 'external');
 
-			if ($serverType === 'internal') {
+			if ($serverType === 'internal' || $serverType === 'spreedbox') {
 				$domain = \OC::$server->getRequest()->getServerHost();
 			} else {
 				$domain = trim($config->getAppValue('ojsxc', 'xmppDomain'));
--- a/lib/Controller/SettingsController.php
+++ b/lib/Controller/SettingsController.php
@@ -47,7 +47,7 @@
 
 		$currentUID = $currentUser->getUID();
 
-		$serverType = $this->getAppValue('serverType');
+		$serverType = $this->getAppValue('serverType', 'spreedbox');
 
 		$data = [
 		 'serverType' => (!empty($serverType))? $serverType : 'internal',
